### PR TITLE
tests: kernel: Use SYS_LOG_INF instead of TC_PRINT

### PR DESCRIPTION
**Pull request by punitvara**
Do not print messages by default on console for test cases.
Use SYS_LOG_INF which provides functionality to choose print whenever
require.

Signed-off-by: Punit Vara <punit.vara@intel.com>

### CODE REVIEW

**Review by tyrantlizard77 — CHANGES_REQUESTED** [review]
NACK
Can you give some context on why we are doing this?
Why do we need these messages suppressed by default?

**Comment by punitvara** [comment]
@andrewboie Output of any test case should be as follow (i.e irq_offload) :
Running test suite test_irq_offload_fn
===================================================================
starting test - test_irq_offload
PASS - test_irq_offload.
===================================================================
===================================================================
PROJECT EXECUTION SUCCESSFUL

Other printk should not be printed unless someone needs it for specific purpose.To fulfill this requirement, either I should remove extra lines or replace with SYS_LOG. As some printk can be important it is not good idea to remove it, SYS_LOG is used to replace those printk or TC_PRINT

**Comment by andrewboie** [comment]
When was this proposed and agreed upon??
Still -1 from me.
@nashif any additional context?

**Comment by nashif** [comment]
> @nashif any additional context?

We need to clean up out of many testcases that have very verbose debug output. Debug output should only be visible when debugging a testcase and using sys_log in this case makes sense. There is no intention to completely remove all output of tests where it shows if the test passes or fails. 

This particular change is not representative and not sure why we only have those 2 files changed but look at XIP output right now:

```
***** BOOTING ZEPHYR OS v1.9.0-rc2 - BUILD: Aug 28 2017 17:20:51 *****
Running test suite test_xip_fn
===================================================================
starting test - test_xip
starting test - kernel_xip
Starting XIP tests
===================================================================
Test globals
PASS - test_xip.
===================================================================
===================================================================
PROJECT EXECUTION SUCCESSFUL
```

There is some duplication here. test_xip could be renamed to rest_globals, so output would be:

```
***** BOOTING ZEPHYR OS v1.9.0-rc2 - BUILD: Aug 28 2017 17:20:51 *****
Running test suite test_xip
===================================================================
starting test - test_globals
PASS - test_globals.
===================================================================
===================================================================
PROJECT EXECUTION SUCCESSFUL
```

File: tests/kernel/irq_offload/src/irq_offload.c [hunk 1]
```diff
@@ -9,6 +9,7 @@
 #include <kernel_structs.h>
 #include <irq_offload.h>
 #include <ztest.h>
+#include <logging/sys_log.h>
```
Comment (nashif) [inline tests/kernel/irq_offload/src/irq_offload.c]: I would include this in ztest.h

File: tests/kernel/xip/src/xip.c [hunk 2]
```diff
@@ -28,13 +29,11 @@ void test_xip(void)
 {
 	int  i;
 
-	TC_START("kernel_xip");
-	PRINT_DATA("Starting XIP tests\n");
-	PRINT_LINE;
+	SYS_LOG_INF("Starting XIP tests\n");
```
Comment (nashif) [inline tests/kernel/xip/src/xip.c]: should be removed, we already announce this using ztest

File: tests/kernel/xip/src/xip.c [hunk 3]
```diff
@@ -28,13 +29,11 @@ void test_xip(void)
...
 
 	/* Test globals are correct */
 
-	TC_PRINT("Test globals\n");
+	SYS_LOG_INF("Test globals\n");
```
Comment (nashif) [inline tests/kernel/xip/src/xip.c]: if we call the test test_globals, then we can get rid of that..

**Review by nashif — CHANGES_REQUESTED** [review]
This is not going to be a scripted change where printk or TC_PRINT are replaced with SYS_LOG globally, please run the tests and examine their output and make the necessary changes per test to achieve a clean output without duplication. The ask here is to improve test output readability, reduce debug output if any etc..

**Comment by punitvara** [comment]
@andrewboie @nashif If you are ok with these changes , I am planning to apply similar changes to other required test cases.

File: tests/kernel/xip/src/xip.c [hunk 4]
```diff
@@ -16,6 +16,7 @@
 #include <tc_util.h>
 #include "test.h"
 #include <ztest.h>
+#include <logging/sys_log.h>
```
Comment (nashif) [inline tests/kernel/xip/src/xip.c]: not needed

**Review by nashif — CHANGES_REQUESTED** [review]

File: tests/kernel/xip/src/xip.c (hunk 4 above)
Comment (punitvara) [reply tests/kernel/xip/src/xip.c]: Done

**Review by nashif — APPROVED** [review]

**Pull request merged**